celements-model/src/main/java/com/celements/store/CelHibernateStore.java

```diff
@@ -128,6 +149,47 @@ public void deleteXWikiDoc(XWikiDocument doc, XWikiContext context) throws XWiki
     }
   }
 
+  public String getDocKey(DocumentReference docRef, String lang) {
+    return getDocKey(Stream.of(serialize(docRef, LOCAL), lang));
+  }
+
+  private String getDocKey(Stream<?> keyParts) {
+    return keyParts.filter(Objects::nonNull).map(Object::toString)
+        .collect(joining(".")).trim();
+  }
+
+  /**
+   * returns a map with all existing docIds for all {@link IdVersion} for the given docRef and lang
+   * sorted by collision count. the map value represents the docKey for the docId.
+   */
+  public SortedMap<Long, String> loadExistingDocKeys(Session session,
+      DocumentReference docRef, String lang) throws HibernateException {
+    Set<Long> allPossibleDocIds = StreamEx.of(docIdComputers)
+        .map(computer -> computer.getDocumentIdIterator(docRef, lang))
+        .flatMap(StreamEx::of)
+        .distinct()
+        .toSet();
+    return loadExistingDocKeys(session, allPossibleDocIds)
+        // docIds for the same doc are sorted by collision count irrespective of their signum
+        // due to 2-complement representation
+        .toSortedMap();
+  }
+
+  @SuppressWarnings("unchecked")
+  private EntryStream<Long, String> loadExistingDocKeys(Session session, Collection<Long> docIds) {

```

Comment:
Didn't know about this wrapper, could have been useful in `UniqueHashIdComputer`, though what would be the advantage of using it here? Our docId-long is neither signed nor unsigned, it's a composition of 3 values (50bit hash | 2bit collision count | 12bit object count). We could write a DocId wrapper for it to represent it OO, though I'm not sure what for since client's don't need to know how the long is encoded.